lib/active_record/connection_adapters/redshift_adapter.rb

```diff
@@ -1,770 +1,9 @@
-        map[Integer] = PG::TextEncoder::Integer.new
-        map[TrueClass] = PG::TextEncoder::Boolean.new
-        map[FalseClass] = PG::TextEncoder::Boolean.new
-        @connection.type_map_for_queries = map
-      end
-
-      def update_typemap_for_default_timezone
-        return if @default_timezone == ActiveRecord.default_timezone || !@timestamp_decoder
-
-        decoder_class =
-          if ActiveRecord.default_timezone == :utc
-            PG::TextDecoder::TimestampUtc
-          else
-            PG::TextDecoder::TimestampWithoutTimeZone
-          end
-
-        @timestamp_decoder = decoder_class.new(@timestamp_decoder.to_h)
-        @connection.type_map_for_results.add_coder(@timestamp_decoder)
-        @default_timezone = ActiveRecord.default_timezone
-
-        # if default timezone has changed, we need to reconfigure the connection
-        # (specifically, the session time zone)
-        configure_connection
-      end
-
-      def add_pg_decoders
-        @default_timezone = nil
-        @timestamp_decoder = nil
-
-        coders_by_name = {
-          'int2' => PG::TextDecoder::Integer,
-          'int4' => PG::TextDecoder::Integer,
-          'int8' => PG::TextDecoder::Integer,
-          'oid' => PG::TextDecoder::Integer,
-          'float4' => PG::TextDecoder::Float,
-          'float8' => PG::TextDecoder::Float,
-          'bool' => PG::TextDecoder::Boolean
-        }
-
-        if defined?(PG::TextDecoder::TimestampUtc)
-          # Use native PG encoders available since pg-1.1
-          coders_by_name['timestamp'] = PG::TextDecoder::TimestampUtc
-          coders_by_name['timestamptz'] = PG::TextDecoder::TimestampWithTimeZone
-        end
-
-        known_coder_types = coders_by_name.keys.map { |n| quote(n) }
-        query = <<~SQL % known_coder_types.join(', ')
-          SELECT t.oid, t.typname
-          FROM pg_type as t
-          WHERE t.typname IN (%s)
-        SQL
-        coders = execute_and_clear(query, 'SCHEMA', []) do |result|
-          result.filter_map { |row| construct_coder(row, coders_by_name[row['typname']]) }
-        end
-
-        map = PG::TypeMapByOid.new
-        coders.each { |coder| map.add_coder(coder) }
-        @connection.type_map_for_results = map
-
-        # extract timestamp decoder for use in update_typemap_for_default_timezone
-        @timestamp_decoder = coders.find { |coder| coder.name == 'timestamp' }
-        update_typemap_for_default_timezone
-      end
-
-      def construct_coder(row, coder_class)
-        return unless coder_class
-
-        coder_class.new(oid: row['oid'].to_i, name: row['typname'])
-      end
-
-      def create_table_definition(*args) # :nodoc:
-        Redshift::TableDefinition.new(self, *args)
-      end
-    end
-  end
-end
+if ActiveRecord.version >= Gem::Version.new('7.1.0')
+  require_relative 'redshift_7_1_adapter'
+elsif ActiveRecord.version >= Gem::Version.new('7.0.0')
+  require_relative 'redshift_7_0_adapter'
+else
+  raise ArgumentError
```

Comment:
I think:

1. This should just be a `RuntimeError` - there's no argument passed to be wrong/invalid
2. We should include a descriptive message to say what's gone wrong